tests/http/server.cpp

```diff
@@ -269,6 +272,17 @@ class Client : public HTTPClient
         const auto& responseHeaders = headers(response);
         const size_t responseHeaderSize =
             std::distance(responseHeaders.begin(), responseHeaders.end());
+
+        // Print headers before error to help debugging
+        if (responseHeaderSize != expectedHeaders.size())
+        {
+            std::cout << "--- Response headers ---" << std::endl;
+            for (const auto& header : responseHeaders)
+                std::cout << header.first << ": " << header.second << std::endl;
+            std::cout << "--- Expected headers ---" << std::endl;
+            for (const auto& header : expectedHeaders)
+                std::cout << header.first << ": " << header.second << std::endl;
+        }
```

Comment:
I want to keep it to improve the life of the next person who has to debug a failing unit test here. Previously all you could see was: "At l.912 3 != 2" which does not help much :-)